pkg/genericapiserver/config.go

```diff
@@ -301,6 +304,10 @@ func (c *Config) ApplyOptions(options *options.ServerRunOptions) *Config {
 			})
 		}
 
+		if secureServingInfo.ServerCert.Generate && secureServingInfo.ServerCA == "" {
```

Comment:
it's also odd for the generate behavior to only trigger when the cert/key args are empty, but to overwrite whatever is specified by a non-empty tls-ca-file arg